infra/terraform/terraform.go

```diff
@@ -379,31 +375,61 @@ func (r *terraform) command(ctx context.Context, args []string, opts ...system.C
 	return out.Bytes(), nil
 }
 
-// serializes terraform vars into given file as JSON
-func (r *terraform) saveVarsJSON(varFile string) error {
-	var config interface{}
-	switch r.Config.CloudProvider {
+// converts a terraform.Config struct into a flat map ready for json serialization
+func configToTerraformVars(cfg Config) (tfvars map[string]interface{}, err error) {
+	tfvars = make(map[string]interface{})
+	switch cfg.CloudProvider {
 	case constants.AWS:
-		config = r.Config.AWS
+		return nil, trace.NotImplemented("aws robotest hasn't been run in a while")
 	case constants.Azure:
-		config = r.Config.Azure
-	case constants.GCE:
-		config = r.Config.GCE
+		return nil, trace.NotImplemented("azure robotest hasn't been run in a while")
```

Comment:
I don't like that I explicitly break aws and azure here, but I haven't yet found the priority to test on either of those platforms, and I'm not aware of anyone who currently uses them.

I can come pencil in some (probably untested) code here if you think it is worthwhile.

Reply:
No, it's fine.